__CONTEXT__

League of Legends is a MOBA (multiplayer online battle arena) where 2 teams (blue and red) face off. There are 3 lanes, a jungle, and 5 roles. The goal is to take down the enemy Nexus to win the game.


__GLOSSARY__

- Warding totem: An item that a player can put on the map to reveal the nearby area. Very useful for map/objectives control.

- Minions: NPC that belong to both teams. They give gold when killed by players.

- Jungle minions: NPC that belong to NO TEAM. They give gold and buffs when killed by players.

- Elite monsters: Monsters with high hp/damage that give a massive bonus (gold/XP/stats) when killed by a team.

- Dragons: Elite monster which gives team bonus when killed. The 4th dragon killed by a team - gives a massive stats bonus. The - - 5th dragon (Elder Dragon) offers a huge advantage to the team.

- Herald: Elite monster which gives stats bonus when killed by the player. It helps to push a lane and destroys structures.

- Towers: Structures you have to destroy to reach the enemy Nexus. They give gold.

- Level: Champion level. Start at 1. Max is 18.


__CONTENT__

This dataset contains the first 10min. stats of approx. 10k ranked games (SOLO QUEUE) from a high ELO (DIAMOND I to MASTER). Players have roughly the same level (ELO).

Each game is unique. The gameId can help you to fetch more attributes from the Riot API.

There are 19 features per team (38 in total) collected after 10min in-game. This includes kills, deaths, gold, experience, level…


In [5]:
import pandas as pd
import numpy as np


In [6]:
# the supplied CSV data file is in the data directory
LoL_data = pd.read_csv('C:\\Users\\luisr\\OneDrive\\Documents\\SpringBoard\\LoLPredictor\\LoLPredictor\\data\\raw\\high_diamond_ranked_10min.csv')

In [7]:
# Summary of the data
LoL_data.info()
LoL_data.shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9879 entries, 0 to 9878
Data columns (total 40 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   gameId                        9879 non-null   int64  
 1   blueWins                      9879 non-null   int64  
 2   blueWardsPlaced               9879 non-null   int64  
 3   blueWardsDestroyed            9879 non-null   int64  
 4   blueFirstBlood                9879 non-null   int64  
 5   blueKills                     9879 non-null   int64  
 6   blueDeaths                    9879 non-null   int64  
 7   blueAssists                   9879 non-null   int64  
 8   blueEliteMonsters             9879 non-null   int64  
 9   blueDragons                   9879 non-null   int64  
 10  blueHeralds                   9879 non-null   int64  
 11  blueTowersDestroyed           9879 non-null   int64  
 12  blueTotalGold                 9879 non-null   int64  
 13  blu

(9879, 40)

In [8]:
# Preview data
pd.options.display.max_columns = None
LoL_data.head(20)

gameId  blueWins  blueWardsPlaced  blueWardsDestroyed  blueFirstBlood  \
0   4519157822         0               28                   2               1   
1   4523371949         0               12                   1               0   
2   4521474530         0               15                   0               0   
3   4524384067         0               43                   1               0   
4   4436033771         0               75                   4               0   
5   4475365709         1               18                   0               0   
6   4493010632         1               18                   3               1   
7   4496759358         0               16                   2               0   
8   4443048030         0               16                   3               0   
9   4509433346         1               13                   1               1   
10  4452162573         0               20                   3               1   
11  4453038156         0               33                   2               1   
12  4515594785         1               18                   1               1   
13  4524924257         0               14                   3               0   
14  4516505202         1               15                   3               1   
15  4482120064         0               17                   1               0   
16  4523758462         1               14                   1               1   
17  4503636905         0               43                   3               0   
18  4486384947         1               21                   4               1   
19  4457103291         0               11                   3               0   

    blueKills  blueDeaths  blueAssists  blueEliteMonsters  blueDragons  \
0           9           6           11                  0            0   
1           5           5            5                  0            0   
2           7          11            4                  1            1   
3           4           5            5                  1            0   
4           6           6            6                  0            0   
5           5           3            6                  1            1   
6           7           6            7                  1            1   
7           5          13            3                  0            0   
8           7           7            8                  0            0   
9           4           5            5                  1            1   
10          4           4            6                  0            0   
11         11          11            7                  1            0   
12          7           1           11                  1            1   
13          4           9            1                  1            0   
14          4           4            4                  0            0   
15          3           7            3                  0            0   
16         10           2            8                  0            0   
17          3           7            3                  1            0   
18          5           4           11                  0            0   
19          5           9            5                  0            0   

    blueHeralds  blueTowersDestroyed  blueTotalGold  blueAvgLevel  \
0             0                    0          17210           6.6   
1             0                    0          14712           6.6   
2             0                    0          16113           6.4   
3             1                    0          15157           7.0   
4             0                    0          16400           7.0   
5             0                    0          15899           7.0   
6             0                    0          16874           6.8   
7             0                    0          15305           6.4   
8             0                    0          16401           7.2   
9             0                    0          15057          

In [9]:
# Copy raw data to analyze and clean
lol_clean = LoL_data.copy()

In [10]:
# Eliminate aggregate columns
# gameId is not a feature in the game, simply an identifier for each match
lol_clean = lol_clean.drop('gameId', axis=1)

# Delete elite moster kills it is an aggregate of two other columns ('blueEliteMonsters' + 'blueHeralds' = blueEliteMonsters)
lol_clean = lol_clean.drop('blueEliteMonsters', axis=1)

# Delete elite moster kills it is an aggregate of two other columns ('redEliteMonsters' + 'redHeralds' = redEliteMonsters)
lol_clean = lol_clean.drop('redEliteMonsters', axis=1)
lol_clean


blueWins  blueWardsPlaced  blueWardsDestroyed  blueFirstBlood  \
0            0               28                   2               1   
1            0               12                   1               0   
2            0               15                   0               0   
3            0               43                   1               0   
4            0               75                   4               0   
...        ...              ...                 ...             ...   
9874         1               17                   2               1   
9875         1               54                   0               0   
9876         0               23                   1               0   
9877         0               14                   4               1   
9878         1               18                   0               1   

      blueKills  blueDeaths  blueAssists  blueDragons  blueHeralds  \
0             9           6           11            0            0   
1             5           5            5            0            0   
2             7          11            4            1            0   
3             4           5            5            0            1   
4             6           6            6            0            0   
...         ...         ...          ...          ...          ...   
9874          7           4            5            1            0   
9875          6           4            8            1            0   
9876          6           7            5            0            0   
9877          2           3            3            1            0   
9878          6           6            5            0            0   

      blueTowersDestroyed  blueTotalGold  blueAvgLevel  blueTotalExperience  \
0                       0          17210           6.6                17039   
1                       0          14712           6.6                16265   
2                       0          16113           6.4                16221   
3                       0          15157           7.0                17954   
4                       0          16400           7.0                18543   
...                   ...            ...           ...                  ...   
9874                    0          17765           7.2                18967   
9875                    0          16238           7.2                19255   
9876                    0          15903           7.0                18032   
9877                    0          14459           6.6                17229   
9878                    0          16266           7.0                17321   

      blueTotalMinionsKilled  blueTotalJungleMinionsKilled  blueGoldDiff  \
0                        195                            36           643   
1                        174                            43         -2908   
2                        186                            46         -1172   
3                        201                            55         -1321   
4                        210                            57         -1004   
...                      ...                           ...           ...   
9874                     211                            69          2519   
9875                     233                            48           782   
9876                     210                            45         -2416   
9877                     224                            48          -839   
9878                     207                            44           927   

      blueExperienceDiff  blueCSPerMin  blueGoldPerMin  redWardsPlaced  \
0                     -8          19.5          1721.0              15   
1                  -1173          17.4          1471.2              12   
2                  -1033          18.6          1611.3              15   
3                     -7          20.1          1515.7              15   
4                    230          21.0          1640.0              17   
...   

In [11]:
# Let's check for missing values by column and mean
missing = pd.concat([lol_clean.isnull().sum(), 100 * lol_clean.mean()], axis=1)
missing.columns=['count', '%']
missing.sort_values(by='count', ascending=False)


count             %
blueWins                          0  4.990384e+01
redWardsPlaced                    0  2.236795e+03
redFirstBlood                     0  4.951918e+01
redKills                          0  6.137666e+02
redDeaths                         0  6.183925e+02
redAssists                        0  6.662112e+02
redDragons                        0  4.130985e+01
redHeralds                        0  1.600364e+01
redTowersDestroyed                0  4.302055e+00
redTotalGold                      0  1.648904e+06
redAvgLevel                       0  6.925316e+02
redTotalExperience                0  1.796173e+06
redTotalMinionsKilled             0  2.173492e+04
redTotalJungleMinionsKilled       0  5.131309e+03
redGoldDiff                       0 -1.441411e+03
redExperienceDiff                 0  3.362031e+03
redCSPerMin                       0  2.173492e+03
redWardsDestroyed                 0  2.723150e+02
blueGoldPerMin                    0  1.650346e+05
blueWardsPlaced                   0  2.228829e+03
blueCSPerMin                      0  2.166996e+03
blueWardsDestroyed                0  2.824881e+02
blueFirstBlood                    0  5.048082e+01
blueKills                         0  6.183925e+02
blueDeaths                        0  6.137666e+02
blueAssists                       0  6.645106e+02
blueDragons                       0  3.619800e+01
blueHeralds                       0  1.879745e+01
blueTowersDestroyed               0  5.142221e+00
blueTotalGold                     0  1.650346e+06
blueAvgLevel                      0  6.916004e+02
blueTotalExperience               0  1.792811e+06
blueTotalMinionsKilled            0  2.166996e+04
blueTotalJungleMinionsKilled      0  5.050967e+03
blueGoldDiff                      0  1.441411e+03
blueExperienceDiff                0 -3.362031e+03
redGoldPerMin                     0  1.648904e+05

Luckly our data shows we have no missing values which we can be confident since that data came from Riots API.

In [12]:
lol_clean.info()
lol_clean.shape


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9879 entries, 0 to 9878
Data columns (total 37 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   blueWins                      9879 non-null   int64  
 1   blueWardsPlaced               9879 non-null   int64  
 2   blueWardsDestroyed            9879 non-null   int64  
 3   blueFirstBlood                9879 non-null   int64  
 4   blueKills                     9879 non-null   int64  
 5   blueDeaths                    9879 non-null   int64  
 6   blueAssists                   9879 non-null   int64  
 7   blueDragons                   9879 non-null   int64  
 8   blueHeralds                   9879 non-null   int64  
 9   blueTowersDestroyed           9879 non-null   int64  
 10  blueTotalGold                 9879 non-null   int64  
 11  blueAvgLevel                  9879 non-null   float64
 12  blueTotalExperience           9879 non-null   int64  
 13  blu

(9879, 37)

In [14]:
# save the data to a new csv file
lol_clean.to_csv('..\\data\\processed\\lol_clean_data', index=False)


__SUMMARY__

This project aims to build a predictive model for 'blueWins' & 'redWins' since these columns are mutually exclusive we will create a model for each side. 

Columns excluded:

- 'gameId'this column is not a feature in the game but simply an identifier

- 'blueEliteMonsters' is an aggregate of two other columns ('blueEliteMonsters' + 'blueHeralds' = blueEliteMonsters)

- 'redEliteMonsters' is an aggregate of two other columns ('redEliteMonsters' + 'rederalds' = redEliteMonsters)

Our original dataset contained 9,879 games with each game containing an in game feature. After our initial observation we deleted 3 columns and 0 rows. Our dataset shows some signs of multicollinearity but futher analysis is require to alter data. 